processor/resourcedetectionprocessor/internal/openshift/openshift.go

```diff
@@ -0,0 +1,84 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package openshift // import "github.com/open-telemetry/opentelemetry-collector-contrib/processor/resourcedetectionprocessor/internal/openshift"
+
+import (
+	"context"
+	"fmt"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	conventions "go.opentelemetry.io/collector/semconv/v1.6.1"
+	"go.uber.org/zap"
+
+	ocp "github.com/open-telemetry/opentelemetry-collector-contrib/internal/metadataproviders/openshift"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/resourcedetectionprocessor/internal"
+)
+
+const (
+	// TypeStr is type of detector.
+	TypeStr = "openshift"
+)
+
+// NewDetector returns a detector which can detect resource attributes on OpenShift 4.
+func NewDetector(set component.ProcessorCreateSettings, dcfg internal.DetectorConfig) (internal.Detector, error) {
+	userCfg := dcfg.(Config)
+
+	if err := userCfg.MergeWithDefaults(); err != nil {
+		return nil, err
+	}
+
+	return &detector{
+		logger:   set.Logger,
+		provider: ocp.NewProvider(userCfg.Address, userCfg.Token),
+	}, nil
+}
+
+type detector struct {
+	logger   *zap.Logger
+	provider ocp.Provider
+}
+
+func (d *detector) Detect(ctx context.Context) (resource pcommon.Resource, schemaURL string, err error) {
+	res := pcommon.NewResource()
+	attrs := res.Attributes()
+
+	infra, err := d.provider.Infrastructure(ctx)
+	if err != nil {
+		return res, "", fmt.Errorf("failed getting Infrastructure metadata: %w", err)
+	}
+
+	// TODO(frzifus): make sure Platform matches the enum.
+	attrs.PutStr(conventions.AttributeCloudProvider, infra.Platform)
+	// TODO(frzifus): make sure Platform matches the enum.
+	attrs.PutStr(conventions.AttributeCloudPlatform, infra.Platform+"_openshift")
+
+	attrs.PutStr(conventions.AttributeK8SClusterName, infra.Name)
+	attrs.PutStr(conventions.AttributeCloudRegion, infra.Region)
+
+	ocpClusterVersion, err := d.provider.OpenShiftClusterVersion(ctx)
+	if err != nil {
+		return res, "", fmt.Errorf("failed getting openshift cluster version: %w", err)
+	}
+	attrs.PutStr("openshift.cluster.version", ocpClusterVersion)
```

Comment:
This also needs to be added to the semantic conventions.

Reply:
We need to wait for:
- https://github.com/open-telemetry/opentelemetry-specification/pull/3016